src/data_source/fetching.rs

```diff
@@ -911,128 +1081,58 @@ where
         // rather than returning `Err` objects, so if there are errors in here they are unexpected
         // and we do want to log them.
         let ts = ts.into_iter().filter_map(ResultExt::ok_or_trace);
+
         // Kick off a fetch for each missing object.
-        let mut fetches = Vec::with_capacity(chunk.len());
+        let mut results = Vec::with_capacity(chunk.len());
         for t in ts {
             // Fetch missing objects that should come before `t`.
-            while chunk.start + fetches.len() < t.height() as usize {
+            while chunk.start + results.len() < t.height() as usize {
```

Comment:
is this while loop because load_range() returns the truncated list? wouldn't this result in lot of unnecessary fetches?

Reply:
nvm  the loop steps ahead because of results.len()